src/main/java/com/base/BaseScheduledFlow.java

```
@@ -0,0 +1,56 @@
+package com.base;
+
+import com.base.processors.ContactProcessor;
+import com.base.processors.DealProcessor;
+import com.getbase.Client;
+import com.getbase.models.*;
+import com.getbase.sync.Sync;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+
+/**
+ * Created by przemek on 19.10.2016.
+ */
+
+@Service
+@Slf4j
+class BaseScheduledFlow {
+
+    private final ContactProcessor contactProcessor;
+    private final DealProcessor dealProcessor;
+    private final Client client;
+
+    @Autowired
+    public BaseScheduledFlow(final Client client, final ContactProcessor contactProcessor, final
+    DealProcessor
```

Strange formatting - why? With multiple parameters constructor (especially with annotations and long names) it looks better to separate it in new line. I would also suggest to use lombok for `@NonNull` check (app fails fast if context beans are messed up) with almost no effort. It would look like this then:

``` java
@Autowired
public BaseScheduledFlow(
        @NonNull final Client client,
        @NonNull final ContactProcessor contactProcessor,
        @NonNull final DealProcessor dealProcessor){
    // (...)
}
```